### PR TITLE
feature-NXBT-2902-fix-release-stage

### PR DESCRIPTION
**Pull request by ataillefer**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -58,8 +49,29 @@ pipeline {
...
       steps {
         container('jx-base') {
-          sh 'jx step git credentials'
-          sh './jx/scripts/release.sh'
+          script {
+            // get release version
+            def version = sh(returnStdout: true, script: 'jx-release-version')
+            echo "Releasing version ${version}"
+
+            // ensure we're not on a detached head
```
Comment (jcarsique) [inline Jenkinsfile]: worth to comment and checkout but there is a clone/checkout option to configure that detached head behavior

File: Jenkinsfile [hunk 2]
```diff
@@ -58,8 +49,29 @@ pipeline {
...
+            sh 'git checkout master'
```
Comment (jcarsique) [inline Jenkinsfile]: this is locking the execution to master branch
the comment is not relevant being in `when branch 'master'` but how is it tested then?

File: Jenkinsfile [hunk 3]
```diff
@@ -58,8 +49,29 @@ pipeline {
...
+            // create the Git credentials
+            sh 'jx step git credentials'
+            sh 'git config credential.helper store'
+
+            // build and push Docker image
+            sh "VERSION=${version} skaffold build -f skaffold.yaml"
+
+            // TODO: check how Skaffold handles cache with two consecutives builds
+            // Ideally, we would only need to tag the ${version} Docker image with the "latest" tag
+            // Something like: docker tag $ORG/$IMAGE_NAME:$VERSION $ORG/$IMAGE_NAME:latest
```
Comment (jcarsique) [inline Jenkinsfile]: +1 
Even more, not using the latest keyword :smile:

File: Jenkinsfile [hunk 4]
```diff
@@ -58,8 +49,33 @@ pipeline {
...
+            echo "Releasing version ${version}"
+
+            sh """
+              #!/usr/bin/env bash -xe
+
+              # ensure we're not on a detached head
+              git checkout master
+
+              # create the Git credentials
+              sh 'jx step git credentials'
```
Comment (jcarsique) [inline Jenkinsfile]: typo

Comment (ataillefer) [reply Jenkinsfile]: Force pushed

**Pull request merged**